pkg/apis/application/v1alpha1/types.go

```diff
@@ -2891,6 +2891,12 @@ func SetK8SConfigDefaults(config *rest.Config) error {
 	config.Timeout = K8sServerSideTimeout
 
 	config.Transport = tr
+	maxRetries := env.ParseInt64FromEnv(utilhttp.EnvRetryMax, 0, 1, math.MaxInt64)
+	if maxRetries > 0 {
+		backoffDurationMS := env.ParseInt64FromEnv(utilhttp.EnvRetryBaseBackoff, 100, 1, math.MaxInt64)
+		backoffDuration := time.Duration(backoffDurationMS) * time.Millisecond
+		config.WrapTransport = utilhttp.WithRetry(maxRetries, backoffDuration)
+	}
```

Comment:
Might be a good idea to document these in env in the docs, for visibility.

Reply:
Agree: I would add to `argocd-cmd-params-cm` configmap and add optional mounts in controller/api server deployments. Example PR https://github.com/argoproj/argo-cd/pull/15646/files

Reply:
done